main.go

```diff
@@ -63,10 +63,14 @@ func main() {
 	// Add flags related to this operator
 	var metricsAddr string
 	var enableLeaderElection bool
+	var collectorImage string
+	var targetAllocatorImage string
 	pflag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	pflag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	pflag.StringVar(&collectorImage, "collector-image", "", "The default OpenTelemetry collector image. This image is used when no image is specified in the CustomResource.")
```

Comment:
Shouldn't this have a default?

Reply:
The default images are hardcoded in the config package https://github.com/open-telemetry/opentelemetry-operator/blob/f7b7b9385c302e5d2f73dfd2700ef3cee6a49506/internal/config/main.go#L74. 

The config package hardcodes image name and sets the tag based on the version.

Reply:
I have made some changes and moved the logic to main. It looks better now and cleaner.